Merge newchem-cpp branch

brittonsmith opened the pull request:
This pull request merges the newchem-cpp branch, which contains several major updates including:

- the new chemical network of PR #177 
- transcription of the Fortran code to C++

For more information, see PR #177 and PR #143.

brittonsmith commented:
@mabruzzo we seem to have some merger conflicts with main now. Are you able to resolve these?

mabruzzo commented:
I’m not at my computer at this moment, but I will check shortly

On Wed, Apr 9, 2025 at 11:52 AM Britton Smith ***@***.***>
wrote:

> @mabruzzo <https://github.com/mabruzzo> we seem to have some merger
> conflicts with main now. Are you able to resolve these?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/grackle-project/grackle/pull/264#issuecomment-2790183883>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG3EHBWVM4SHJ5ROL445A4D2YU63DAVCNFSM6AAAAABZH6TIOGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJQGE4DGOBYGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
> *brittonsmith* left a comment (grackle-project/grackle#264)
> <https://github.com/grackle-project/grackle/pull/264#issuecomment-2790183883>
>
> @mabruzzo <https://github.com/mabruzzo> we seem to have some merger
> conflicts with main now. Are you able to resolve these?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/grackle-project/grackle/pull/264#issuecomment-2790183883>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG3EHBWVM4SHJ5ROL445A4D2YU63DAVCNFSM6AAAAABZH6TIOGVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDOOJQGE4DGOBYGM>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

mabruzzo commented:
That force push just happened because I accidentally pushed to this branch 2 seconds earlier

brittonsmith commented on src/clib/solve_rate_cool_g.F:
```diff
@@ -329,7 +556,14 @@ subroutine solve_rate_cool_g(icool, d, e, u, v, w, de,
+      endif
+
+!     ignore metal chemistry/cooling below this metallicity
+      min_metallicity = 1.d-9 / z_solar
```
In solar units, this is a metallicity of just under 1e-7 Zsun. I suspect I've done this wrong and that the min metallicity is meant to be 1e-9 Zsun. This would be consistent with other places in the code where this is done by hand.

brittonsmith commented on src/clib/solve_rate_cool_g.F:
```diff
@@ -1891,29 +3254,47 @@ subroutine rate_timestep_g(
      &               +  k25shield(i)*HeII(i,j,k)/4._DKIND
      &               +  k26shield(i)*HeI(i,j,k)/4._DKIND)
 
+!     HII, HeII, HeIII recombination heating
+
+            edot(i) = edot(i) - chunit * (
+     &    13.6_DKIND*( k1(i) * HI(i,j,k)   * de(i,j,k)
+     &               - k2(i) * HII(i,j,k)  * de(i,j,k)          )
+     &  + 24.6_DKIND*( k3(i) * HeI(i,j,k)  * de(i,j,k)/4._DKIND
+     &               - k4(i) * HeII(i,j,k) * de(i,j,k)/4._DKIND )
+     &  + 79.0_DKIND*( k5(i) * HeII(i,j,k) * de(i,j,k)/4._DKIND
+     &               - k6(i) * HeIII(i,j,k)* de(i,j,k)/4._DKIND )
+     &            )
+
```
@gregbryan, @jwise77  I would really like to get your thoughts on this block of code here. With this, we are now adding a heating term proportional to the recombination rate minus the collisional ionization rate. This sort of cancels out the conventional recombination cooling term in cool1d_multi_g.F. I think the assumption here is that recombination photons are being reabsorbed. Does that sound right? The main result is a slightly higher (~20%) equilibrium temperature when a uv background is present. Does this seem like a reasonable addition?

The second thing to point out is that this term is not being applied in cool1d_mult, but in solve_rate_cool near to where chemical heating from H2 formation is being applied. This basically means that the cooling rate applied during solve_rate_cool is inconsistent with what one would derive by calling calculate_cooling_time. I assume this was done here as it relies on the k-rate values, which are not being passed into cool1d_multi. If this is a term worth keeping, do you think it's worth moving into cool1d_multi properly?

brittonsmith commented on src/clib/calc_tdust_1d_g.F:
```diff
@@ -344,35 +373,52 @@ subroutine calc_kappa_gr_g(
 
-      real*8 kgr1, kgr200
-      parameter(kgr1 = 4.0e-4_DKIND, kgr200 = 16.0_DKIND)
+      real*8, parameter :: kgr1 = 4.0e-4_DKIND / 0.009387d0
+      real*8, parameter :: kgr200 = 16.0_DKIND / 0.009387d0
```
I concur that this normalization needs to be there. Specifically, equation 17 of Omukai (2000) gives kappa **at Z = Z_local**. However, just below that this paper seems define dust-to-gas ratio at Z_local as 0.934e-2 whereas the value used here is from Pollack (1994). As well, the T^2 proportionality used here is taken from Dopcke et al. (2011), who have a slightly different normalization (by about a factor of 2). This at least needs to be pointed out here.

mabruzzo commented:
Oops -- I accidentally pushed changed directly onto this branch. I caught it super quickly and stripped them right back off.

(Sorry about that -- It's been a while since the last time I did this)

mabruzzo committed:
Ensure that consistent ln(T) vals are used for rate calculations

mabruzzo committed:
update the recorded property values

mabruzzo committed:
slightly simplify cool_multi_time

mabruzzo committed:
remove the definition of NULL macro

mabruzzo committed:
rename (and cppify) grackle_units.c -> api/units.cpp

mabruzzo committed:
remove some unnecessary include-directives

ChristopherBignamini committed:
Function signature update

ChristopherBignamini committed:
Call to calc_kappa_gr_g updated according to the new function signature

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ChristopherBignamini committed:
const condition added to input parameters and local variables

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ChristopherBignamini committed:
calc_kappa_gr function/files renaming

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

mabruzzo committed:
compute electron density earlier in cool1d_multi_g

mabruzzo committed:
factor out metallicity and electron-density calculation logic

mabruzzo committed:
rm myde from Cool1DMultiScratchBuf

It is now temporarily allocated inside of cool1d_multi_g (the plan is to
pass in a buffer as an argument). The new temporary variable is now
called `nelec_times_mH`

mabruzzo committed:
allocate nelec_times_mH outside of cool1d_multi_g

ChristopherBignamini committed:
File renaming

ChristopherBignamini committed:
alsp primitive data type updated to double

ChristopherBignamini committed:
Single element array variable changed into scalars

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

brittonsmith committed:
Merge with newchem-cpp.

brittonsmith committed:
Merge pull request #496 from mabruzzo/ncc/excise-PhotoRxnRateCollection

[newchem-cpp] Remove `PhotoRxnRateCollection`

brittonsmith committed:
Merge pull request #522 from mabruzzo/ncc/rm-SimpleVec

[newchem-cpp] Delete `SimpleVec`

brittonsmith committed:
Merge pull request #518 from ChristopherBignamini/newchem-cpp_transcribe_calc_kappa_gr_g

[newchem-cpp] Transcribe calc_kappa_gr_g

brittonsmith committed:
Merge with newchem-cpp.

brittonsmith committed:
Merge pull request #523 from mabruzzo/ncc-cleanup

Light cleanup related to `calc_temp_cloudy_g`, `cool_multi_time_g`, `calc_tdust_3d_g`

brittonsmith committed:
Merge branch 'newchem-cpp' into ncc/cppify-internal_units

ChristopherBignamini committed:
Merge remote-tracking branch 'upstream/newchem-cpp' into newchem-cpp_transcribe_calc_gr_balance_g

brittonsmith committed:
Merge pull request #524 from mabruzzo/ncc/cppify-internal_units

Cppify `internal_units.h`

brittonsmith committed:
Replace approximate atomic mass with mass number.

brittonsmith committed:
Add comment to distinguish between hydrogen and deuterium.

brittonsmith committed:
Use correct metal field for scaling nuclides.

brittonsmith committed:
Remove unused import.

brittonsmith committed:
Merging with newchem-cpp.

brittonsmith committed:
Merge with newchem-cpp.

brittonsmith committed:
Merge pull request #538 from mabruzzo/ncc/eliminate-POW

[newchem-cpp] Eliminate a bunch of macros from `grackle_macros.h`

mabruzzo committed:
Merge branch 'newchem-cpp' into ncc/reorganize-tests

mabruzzo committed:
Merge branch 'newchem-cpp' into ncc/reorganize-tests

mabruzzo committed:
Merge branch 'newchem-cpp' into ncc/stop-passing-p2d-buffer

mabruzzo committed:
Merge branch 'ncc/stop-passing-p2d-buffer' into ncc/gas-props2

brittonsmith committed:
Merge pull request #509 from mabruzzo/ncc/grtestutils_better_assert

[newchem-cpp] Improve Array-Comparison Assertion

brittonsmith committed:
Merge pull request #513 from mabruzzo/ncc/dependency-install-logic

[newchem-cpp] Shift CI dependency installation logic to `install_dependencies.py`

mabruzzo committed:
Merge branch 'newchem-cpp' into ncc/stop-passing-p2d-buffer

mabruzzo committed:
Merge branch 'ncc/stop-passing-p2d-buffer' into ncc/gas-props2

mabruzzo committed:
Merge branch 'ncc/gas-props2' into ncc/excise-p2d

mabruzzo committed:
Merge branch 'ncc/gas-props2' into ncc/reorg-tabulated

mabruzzo committed:
Merge branch 'ncc/reorg-tabulated' into ncc/calculate_temperature-basic_gas_props

mabruzzo committed:
fix an oversight

mabruzzo committed:
Merge branch 'ncc/reorg-tabulated' into ncc/light-tabulated-refactor

mabruzzo committed:
Merge branch 'ncc/reorg-tabulated' into ncc/LnTPreparer

mabruzzo committed:
Merge pull request #512 from mabruzzo/ncc/reorganize-tests

Ncc/reorganize tests

mabruzzo committed:
Merge branch 'newchem-cpp' into ncc/stop-passing-p2d-buffer

mabruzzo committed:
Merge branch 'ncc/stop-passing-p2d-buffer' into ncc/gas-props2

mabruzzo committed:
Merge branch 'ncc/gas-props2' into ncc/reorg-tabulated

mabruzzo committed:
Merge branch 'ncc/reorg-tabulated' into ncc/calculate_temperature-basic_gas_props

mabruzzo committed:
Merge branch 'ncc/gas-props2' into ncc/excise-p2d

mabruzzo committed:
Merge branch 'ncc/reorg-tabulated' into ncc/light-tabulated-refactor

mabruzzo committed:
Merge branch 'ncc/reorg-tabulated' into ncc/LnTPreparer

mabruzzo committed:
Updated recorded cooling_time

It turns out that the value computed on CI and on my personal machine
were -4031685600248.466 and -4031685600261.704. These values had a
relative difference of ~ -3.3e-12, which slightly exceeded our tolerance
(2e-12). Consequently, we just averaged the values

mabruzzo committed:
Merge branch 'newchem-cpp' into ncc/cppify-units

mabruzzo committed:
convert rate_functions to C++

mabruzzo committed:
rename _k13dd_rate -> k13dd_rate_ (for C++ compatability)

For context, C++ reserves all names starting with an underscore.
While I was there, I also wrote a short docstring (and made my_chemistry
into a pointer-to-const)

mabruzzo committed:
adjust format of the heading of rate_functions.c

mabruzzo committed:
use GR_INTERNAL_ERROR within rate_functions

Basically, we replaced every fprintf(stderr, ...) followed by a call to
exit or abort with GR_INTERNAL_ERROR. (I had no idea we were aggressive
about crashing the program)

mabruzzo committed:
Prefix all math functions with std::

mabruzzo committed:
move k13dd_rate_ into internal namespace

mabruzzo committed:
adjust the included headers

mabruzzo committed:
convert `cie_thin_cooling_rate_tables.{h->hpp}`

mabruzzo committed:
more properly the inline header definitions of t_cie_c and cie_table_c

mabruzzo committed:
adjust heading of status_reporting.[ch]

mabruzzo committed:
rename (& C++-ify) status_reporting.[ch] -> support/status_reporting.[ch]pp

mabruzzo committed:
status_reporting.cpp: use C++ versions of headers

mabruzzo committed:
status_reporting.cpp: replace NULL to nullptr

mabruzzo committed:
status_reporting.cpp: replace new & delete with std::vector

mabruzzo committed:
status_reporting.hpp: C++-ify the attributes

mabruzzo committed:
status_reporting: get rid of extern "C"

mabruzzo committed:
status_reporting: mv struct & functions into internal namespace

mabruzzo committed:
remove duplicate definition of GR_INTERNAL_UNREACHABLE_ERROR

mabruzzo committed:
rename macros to remove double-underscore (bad practice in C++)

mabruzzo committed:
mv test_status_reporting to support subdirectory

mabruzzo committed:
adjust overview-dcostring in status_reporting so that doxygen properly associates it with the machinery declared in the file

mabruzzo committed:
fix a minor bug (that I accidentally introduced)

mabruzzo committed:
adjust the Makefile

mabruzzo committed:
rename & C++ify index_helper.[ch] -> support/index_helper.[ch]pp

mabruzzo committed:
get rid of the C-style struct typedefs

mabruzzo committed:
remove `static` specifier from `inline` functions

mabruzzo committed:
Convert build_index_helper_ to an inline function

The main motivation for doing this was to get rid of the
index_helper.cpp file. We can always revert this choice later

mabruzzo committed:
index_helper: stop declaring functions with C linkage

mabruzzo committed:
Modernize the heading of index_helper.hpp file

mabruzzo committed:
directly include the grackle.h header file

mabruzzo committed:
get rid of some unnecessary comment blocks

mabruzzo committed:
adjust docstrings

mabruzzo committed:
introduce tests for index_helper

mabruzzo committed:
rename grackle_index_helper & field_flat_index_range

This commit also puts the structs & functions into the internal Grackle
namespace

mabruzzo committed:
start using clang-format to format index_helper.hpp

mabruzzo committed:
address a compiler warning

mabruzzo committed:
compute electrong density and metallicity outside of cool1d_multi_g

mabruzzo committed:
calc_temp1d_cloudy: adjust my_chemistry and my_fields so that they are pointers to consts

mabruzzo committed:
adjust View declarations

mabruzzo committed:
basic_gas_props: better arg const-correctness

This commit makes my_chemistry, primordial_cloudy_data, my_fields into
pointers to const values (rather than raw pointers)

mabruzzo committed:
Merge branch 'ncc/minor-patch-calc_temp1d_cloudy' into ncc/calculate_temperature-basic_gas_props

mabruzzo committed:
make calc_temp_cloudy accept pointer to const chemistry_data

mabruzzo committed:
Merge branch 'ncc/minor-patch-calc_temp1d_cloudy' into ncc/LnTPreparer

mabruzzo committed:
move the declaration of LogTLinInterpScratchBuf to the lnT_prep.hpp header

mabruzzo committed:
apply clang-format on LogTLinInterpScratchBuf

mabruzzo committed:
update docstring for LogTLinInterpScratchBuf

mabruzzo committed:
rename LogTLinInterpScratchBuf -> LnTLinInterpBuf

mabruzzo committed:
Make a minor tweak (that may slightly impact gold-standard)

This tweak will help with the next PR in the sequence

mabruzzo committed:
create the extended_gas_props function

mabruzzo committed:
add a comment describing the nelec_times_mH bufferr

mabruzzo committed:
make metallicity arg a `const double*`

ChristopherBignamini committed:
Merge remote-tracking branch 'upstream/newchem-cpp' into newchem-cpp_transcribe_solve_cubic_equation

ChristopherBignamini committed:
Merge remote-tracking branch 'upstream/newchem-cpp' into newchem-cpp_transcribe_calc_grain_size_increment_species_1d

brittonsmith committed:
Add more discussion of uncertainty around mass densities.

brittonsmith committed:
Update gold standard.

brittonsmith committed:
Skip initializing ions for elements we are not carrying.

brittonsmith committed:
Add exclusion for primordial_chemistry = 0 and dust_recombination_cooling = 1.

brittonsmith committed:
Merge pull request #528 from mabruzzo/ncc/stop-passing-p2d-buffer

[newchem-cpp] stop passing `p2d` buffer

mabruzzo committed:
Merge pull request #505 from mabruzzo/ncc/improve-cmake

[newchem-cpp] Assorted CMake Improvements

brittonsmith committed:
Rename elements property as nuclides.

brittonsmith committed:
Rename fname to field_name and fix warning in a regex.

brittonsmith committed:
Merge pull request #536 from brittonsmith/ncc-fluid-inj-fields

[newchem-cpp] Add FluidContainer support for setting nuclides/grains from injection metals

ChristopherBignamini committed:
Merge remote-tracking branch 'upstream/newchem-cpp' into newchem-cpp_transcribe_calc_grain_size_increment_species_1d

ChristopherBignamini committed:
Merge remote-tracking branch 'upstream/newchem-cpp' into newchem-cpp_transcribe_calc_gr_balance_g

ChristopherBignamini committed:
Merge remote-tracking branch 'upstream/newchem-cpp' into newchem-cpp_transcribe_solve_cubic_equation

ChristopherBignamini committed:
Debugging

brittonsmith committed:
Merge pull request #539 from mabruzzo/ncc/cppify-units

[newchem-cpp] renames (and C++-ify) `grackle_units.c` -> `api/units.cpp`

brittonsmith committed:
Merge pull request #540 from mabruzzo/ncc/cppify-rate_functions

[newchem-cpp] C++-ify `rate_functions.c` and `cie_thin_cooling_rate_tables.h`

brittonsmith committed:
Merge pull request #541 from mabruzzo/ncc/cppify-status_reporting

[newchem-cpp] rename (& C++-ify) `status_reporting.[ch]` -> `support/status_reporting.[ch]pp`

brittonsmith committed:
Merge with newchem-cpp.

brittonsmith committed:
Make separate step to generate gold standard.

brittonsmith committed:
Merge pull request #542 from mabruzzo/ncc/cppify-index_helper

[newchem-cpp] rename and C++-ify `index_helper.h` -> `support/index_helper.hpp`

brittonsmith committed:
Merge pull request #529 from mabruzzo/ncc/gas-props2

[newchem-cpp] Factor out gas property calculation

mabruzzo committed:
minor bugfix

brittonsmith committed:
Merge pull request #550 from mabruzzo/ncc/minor-bugfix

[newchem-cpp] Introduce a minor bugfix

brittonsmith committed:
Merge with newchem-cpp.

brittonsmith committed:
Merge pull request #530 from mabruzzo/ncc/excise-p2d

[newchem-cpp] Entirely remove the `p2d` array

brittonsmith committed:
Merge with newchem-cpp.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

brittonsmith committed:
Merge pull request #533 from mabruzzo/ncc/reorg-tabulated

[newchem-cpp] Reorganize Tabulated Routines

brittonsmith committed:
Merge pull request #543 from mabruzzo/ncc/minor-patch-calc_temp1d_cloudy

[newchem-cpp] Minor tweaks to calc_temp1d_cloudy and basic_gas_props

brittonsmith committed:
Hopeful merge with newchem-cpp.

brittonsmith committed:
Merge pull request #531 from mabruzzo/ncc/calculate_temperature-basic_gas_props

[newchem-cpp] `calc_temp_cloudy`: Refactor to call `basic_gas_props`

brittonsmith committed:
Merge pull request #534 from mabruzzo/ncc/light-tabulated-refactor

[newchem-cpp] tabulated refactor

brittonsmith committed:
Merge branch 'newchem-cpp' into ncc/LnTPreparer

mabruzzo committed:
Merge pull request #548 from brittonsmith/ncc/make-gold

[newchem-cpp] Make separate step to generate gold standard.

ChristopherBignamini committed:
Merge remote-tracking branch 'upstream/newchem-cpp' into newchem-cpp_transcribe_calc_grain_size_increment_species_1d

ChristopherBignamini committed:
alsp and alsp_data renamed as kappa and kappa_data

ChristopherBignamini committed:
SN_metal_data renamed as selected_inj_path_metal_densities and header include update

ChristopherBignamini committed:
nSN renamed as n_selected_inj_paths

ChristopherBignamini committed:
Documentation updated

ChristopherBignamini committed:
Documentation update

ChristopherBignamini committed:
const keyword added

ChristopherBignamini committed:
sgsp renamed as sigma_per_gas_mass

ChristopherBignamini committed:
ssp renamed as bulk_density

ChristopherBignamini committed:
SN_kp0sp_data renamed as opac_coef_table_data

ChristopherBignamini committed:
const keyword added

mabruzzo committed:
move a handful of macro definitions from utils-cpp.hpp into support/config.hpp

ChristopherBignamini committed:
File moved

ChristopherBignamini committed:
File moved

ChristopherBignamini committed:
File path update

ChristopherBignamini committed:
Include path update

mabruzzo committed:
transfer implementation of View into its own file

clang-format adjusted formatting

mabruzzo committed:
clean up ordering of files in CMakeLists.txt

mabruzzo committed:
add some basic unit tests for GRIMPL_NS::View

ChristopherBignamini committed:
SN0_N renamed as n_inj_pathways

ChristopherBignamini committed:
in, jn, kn replaced with a 3 element array

ChristopherBignamini committed:
in, jn, kn replaced with a 3 element array

ChristopherBignamini committed:
Old declaration removed

ChristopherBignamini committed:
Cleanup and adoption of new calc_grain_size_increment_species_1d signature

ChristopherBignamini committed:
gr_Size removed from function signature and internally calculated

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

mabruzzo committed:
add support for implicitly casting a View to a const version

brittonsmith committed:
Fix path for include.

brittonsmith committed:
Update gold standard for PR #535.

brittonsmith committed:
Merge pull request #535 from mabruzzo/ncc/LnTPreparer

Refactor logT precalculation

brittonsmith committed:
Merge with newchem-cpp.

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

brittonsmith committed:
Update gold standard for PR #537.

brittonsmith committed:
Merge branch 'ncc/unify-LnT-calculation' of github.com:mabruzzo/grackle into ncc/unify-LnT-calculation

brittonsmith committed:
Merge pull request #537 from mabruzzo/ncc/unify-LnT-calculation

[newchem-cpp] Unify/Simplify ln(T) calculations

brittonsmith committed:
Merge branch 'newchem-cpp' into ncc/extended-gas-props

brittonsmith committed:
Merge pull request #545 from mabruzzo/ncc/extended-gas-props

[newchem-cpp] Factor out metallicity and electron density calculations from `cool1d_multi_g`

ChristopherBignamini committed:
Documentation updated

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ChristopherBignamini committed:
Merge remote-tracking branch 'upstream/newchem-cpp' into newchem-cpp_transcribe_calc_grain_size_increment_species_1d

ChristopherBignamini committed:
Merge branch 'newchem-cpp_transcribe_calc_grain_size_increment_species_1d' of https://github.com/ChristopherBignamini/grackle into newchem-cpp_transcribe_calc_grain_size_increment_species_1d

ChristopherBignamini committed:
Merge remote-tracking branch 'upstream/newchem-cpp' into newchem-cpp_transcribe_calc_gr_balance_g

ChristopherBignamini committed:
Merge branch 'newchem-cpp_transcribe_calc_grain_size_increment_species_1d' into newchem-cpp_transcribe_calc_gr_balance_g

ChristopherBignamini committed:
calc_grain_size_increment_species_1d fortran implementation removed

ChristopherBignamini committed:
Fortran implementation removed

ChristopherBignamini committed:
Include fix

ChristopherBignamini committed:
Fortran declaration removed

ChristopherBignamini committed:
Wrapper removed

ChristopherBignamini committed:
Old source file removed

ChristopherBignamini committed:
Merge remote-tracking branch 'upstream/newchem-cpp' into newchem-cpp_transcribe_solve_cubic_equation

ChristopherBignamini committed:
Merge branch 'newchem-cpp_transcribe_calc_gr_balance_g' into newchem-cpp_transcribe_solve_cubic_equation

ChristopherBignamini committed:
Old declaration removed

ChristopherBignamini committed:
Include cleanup

ChristopherBignamini committed:
Namespace update

ChristopherBignamini committed:
Old fortran implementation removed

ChristopherBignamini committed:
Namespace usage updated

ChristopherBignamini committed:
const condition added, variable renaming, todo added and namespace usage update

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ChristopherBignamini committed:
Namespace update

ChristopherBignamini committed:
Commented calls to old fortran version of calc_gr_balance_g removed

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

ChristopherBignamini committed:
Merge branch 'newchem-cpp_transcribe_calc_grain_size_increment_species_1d' into newchem-cpp_transcribe_solve_cubic_equation

ChristopherBignamini committed:
Namespace update

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

brittonsmith committed:
Merge pull request #507 from ChristopherBignamini/newchem-cpp_transcribe_calc_grain_size_increment_species_1d

[newchem-cpp] transcribe calc_grain_size_increment_species_1d

brittonsmith committed:
Update gold standard for PR #507.

brittonsmith committed:
Merge pull request #556 from brittonsmith/nccv11

Update gold standard for PR #507.

brittonsmith committed:
Merge pull request #519 from ChristopherBignamini/newchem-cpp_transcribe_calc_gr_balance_g

[newchem-cpp] Transcribe calc_gr_balance_g

brittonsmith committed:
Update gold standard for PR #519.

brittonsmith committed:
Merge pull request #557 from brittonsmith/nccv12

Update gold standard for PR #519.

brittonsmith committed:
Merge pull request #520 from ChristopherBignamini/newchem-cpp_transcribe_solve_cubic_equation

[newchem-cpp] Transcribe solve_cubic_equation

mabruzzo committed:
Merge pull request #552 from mabruzzo/ncc/mv-view

[newchem-cpp] Move definition of the View and add some tests

mabruzzo committed:
fix a few minor oversights in src/clib/time_deriv_0d.hpp

mabruzzo committed:
time_deriv_0d: adjust the way namespace is declared

mabruzzo committed:
time_deriv_0d: add boiler-plate heading

mabruzzo committed:
introduce the MAX_EVOLVED_SPECIES_FIELDS constant

mabruzzo committed:
stop tracking iter in time_deriv_0d::FrozenSimpleArgs

mabruzzo committed:
step_rate_newton_raphson: adjust boiler-plate heading

mabruzzo committed:
step_rate_newton_raphson: make some surface level tweaks

mabruzzo committed:
step_rate_newton_raphson: start using MAX_EVOLVED_SPECIES_FIELDS

mabruzzo committed:
step_rate_newton_raphson: const-qualify some of the arguments

mabruzzo committed:
`time_deriv_0d`: pre-allocate evolved fields in 1-zone grackle_field_data

Currently, `time_deriv_0d` works by preparing a context pack with all of
the infrastructure for creating a 1-element version of grackle_field_data.
This was a pragmatic choice early during transcription, but it's hacky &
inefficient.

Before this PR:

- the `time_derivative_0d::derivative` function would overwrite
  `pack.field.internal_energy` and `pack.field.e_density`,
  `pack.field.HI_density`, `pack.field.HII_density`, ... to point to the
  buffers provided in the function arguments
- now, the context pack will pre-allocate the evolved buffers tracked
  within the 1-element version `grackle_field_data` and we will just copy over
  the values from the function arguments

Aside: for all fields that aren't evolved (e.g. `density`, `dust_density`,
`volumetric_heating_rate`, RT-fields, etc.) the 1-zone `grackle_field_data`
instance just points to the appropriate location in the full 3D field.

mabruzzo committed:
fix a minor issue in the Makefile

mabruzzo committed:
another Makefile tweak

mabruzzo committed:
introduce opaque_storage destructor

brittonsmith committed:
Merge pull request #559 from mabruzzo/ncc/Makefile-hotfix

[newchem-cpp] Makefile hotfix: remove old-filenames from Makefile

brittonsmith committed:
Merge pull request #553 from mabruzzo/ncc/view-const-cast

[newchem-cpp] Add support for const-casting

brittonsmith committed:
Merge pull request #558 from mabruzzo/ncc/tderiv_tweaks

[newchem-cpp] `time_deriv_0d`: misc tweaks & pre-allocate evolved fields in 1-zone `grackle_field_data`

mabruzzo committed:
address minor oversight

brittonsmith committed:
Merge pull request #563 from mabruzzo/ncc/opaque_storage_destructor

[newchem-cpp] Introduce destructor to `gr_opaque_storage`

brittonsmith committed:
Merge with newchem-cpp and fix a few things.

brittonsmith committed:
Formatting.

brittonsmith committed:
typo

brittonsmith committed:
Remove O density variables.

brittonsmith committed:
Revert change to interaction page.

brittonsmith committed:
remove blank lines.

brittonsmith committed:
Merge pull request #364 from KeiganM/networkz

[newchem-cpp] extended reaction table

brittonsmith committed:
Remove dust_sublimation parameter and functionality.

brittonsmith committed:
Merge pull request #566 from brittonsmith/ncc-remove-sublimation

Remove dust_sublimation parameter and functionality.

brittonsmith committed:
Revert "Remove dust_sublimation parameter and functionality."

brittonsmith committed:
Merge pull request #568 from grackle-project/revert-566-ncc-remove-sublimation

Revert "Remove dust_sublimation parameter and functionality."

ChristopherBignamini committed:
Added check to avoid dereferencing of null pointers in logging

mabruzzo committed:
Merge pull request #572 from ChristopherBignamini/radiative_transfer_data_logging_fix

[newchem-cpp] Added check to avoid dereferencing of null pointers in logging
